src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookMetadataProvider.cs

```diff
@@ -144,168 +153,213 @@ private void Apply(ActionModel action)
-            IReadOnlyList<IWebHookBodyTypeMetadataService> bodyTypeMetadata)
+        /// <remarks>
+        /// <paramref name="receiverBodyTypeMetadata"/> is <see langword="null"/> only if
+        /// <paramref name="actionBodyTypeMetadata"/> is a <see cref="GeneralWebHookAttribute"/> instance.
+        /// </remarks>
+        protected void EnsureValidBodyTypeMetadata(
+            IWebHookBodyTypeMetadata actionBodyTypeMetadata,
+            IWebHookBodyTypeMetadataService receiverBodyTypeMetadata)
         {
-            if (eventFromBodyMetadata == null)
+            if (actionBodyTypeMetadata == null)
             {
-                throw new ArgumentNullException(nameof(eventFromBodyMetadata));
+                throw new ArgumentNullException(nameof(actionBodyTypeMetadata));
             }
-            if (bodyTypeMetadata == null)
+
+            // Confirm the attribute's BodyType is valid on its own. Avoid Enum.IsDefined because we want to
+            // distinguish invalid flag combinations from undefined flags. This method is not called if
+            // actionBodyTypeMetadata.BodyType is AllBodyTypes.
+            switch (actionBodyTypeMetadata.BodyType)
             {
-                throw new ArgumentNullException(nameof(bodyTypeMetadata));
+                case WebHookBodyType.Form:
+                case WebHookBodyType.Json:
+                case WebHookBodyType.Xml:
+                    // Just right.
+                    break;
+
+                case 0:
+                case WebHookBodyType.Form | WebHookBodyType.Json:
+                case WebHookBodyType.Form | WebHookBodyType.Xml:
+                case WebHookBodyType.Json | WebHookBodyType.Xml:
+                    {
+                        // 0 or contains an invalid combination of flags.
+                        var message = string.Format(
+                            CultureInfo.CurrentCulture,
+                            Resources.MetadataProvider_InvalidAttributeBodyType,
+                            actionBodyTypeMetadata.GetType(),
+                            typeof(IWebHookBodyTypeMetadata),
+                            nameof(IWebHookBodyTypeMetadata.BodyType),
+                            actionBodyTypeMetadata.BodyType,
+                            WebHookConstants.AllBodyTypes,
+                            nameof(WebHookBodyType));
+                        throw new InvalidOperationException(message);
+                    }
+
+                default:
+                    {
+                        // Value contains undefined flags.
+                        var message = string.Format(
+                            CultureInfo.CurrentCulture,
+                            Resources.General_InvalidEnumValue,
+                            typeof(WebHookBodyType),
+                            actionBodyTypeMetadata.BodyType);
+                        throw new InvalidOperationException(message);
+                    }
```

Comment:
Do we really need all of this? This seem like it's really doing a ton of work to tell folks they did something silly.

Reply:
Commenting again after reading through the rest of the changes. About 60% of the code seems dedicated to detecting and reporting invalid cases. 

Is there a way we could simplify the contract that has to be implemented by the runtime - IE check once when actions are discovered, and then ignore/crash on invalid scenarios that can't be detected at startup?

Reply:
Discussed offline and I'll switch to support `[GeneralWebHook]` without resorting to a `[Flags]` enum.